src/Swarm/Game/ScenarioInfo.hs

```diff
@@ -289,9 +289,15 @@ loadScenarioDir em dir = do
  where
   -- Keep only files which are .yaml files or directories that start
   -- with something other than an underscore.
-  keepYamlOrPublicDirectory =
-    filter
-      (\f -> takeExtensions f == ".yaml" || (takeExtensions f == "" && head f /= '_'))
+  keepYamlOrPublicDirectory = filterM . isCatalogEntry
+
+  -- Whether the directory entry should be included in the scenario catalog.
+  isCatalogEntry d f = do
+    isDir <- doesDirectoryExist $ d </> f
+    return $
+      if isDir
+        then not $ "_" `isPrefixOf` f
+        else takeExtensions f == ".yaml"
```

Comment:
The description does not seem to match filtering YAML-only files. 😕

Reply:
Do you mean the comment?

    -- Whether the directory entry should be included in the scenario catalog.

A directory "entry" is anything output by the `listDirectory` function.  So "directory" in that phrasing is referring to the argument to the `listDirectory` function.  But I recognize the ambiguity, and have re-worded the comment to:

    -- Whether the directory or file should be included in the scenario catalog.